packages/playwright-core/src/server/injected/injectedScript.ts

```
@@ -1398,7 +1398,10 @@ export class InjectedScript {
           return { received: null, matches: false };
         received = value;
       } else if (expression === 'to.have.class') {
-        received = element.classList.toString();
+        return {
```

Let's throw a nice error when `!options.expectedText`, just in case?